infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogEntrySender.java

```diff
@@ -148,8 +148,15 @@ public void send(UUID logEntrySyncEventId) {
             }
         }
 
-        logReplicationFSM.input(new LogReplicationEvent(LogReplicationEvent.LogReplicationEventType.LOG_ENTRY_SYNC_CONTINUE,
-                new LogReplicationEventMetadata(logEntrySyncEventId), logReplicationFSM));
+        if (isRetry && !taskActive) {
```

Comment:
what about the case of initial send() where retry = false and no data found(task active = false).  Shouldnt there be a delay in that case?